src/vivarium/framework/dependency.py

```diff
@@ -0,0 +1,124 @@
+"""
+=================================
+Dependency Management in Vivarium
+=================================
+
+This module provides a tool to manage dependencies on resources within a
+``Vivarium`` simulation. These resources take the form of things that can be
+created and utilized by components, for example columns in the state table or
+named value pipelines. Because these need to be created before they can be used,
+they are sensitive to ordering. The intent behind this tool is to provide an
+interface that allows other managers to register resources with the dependency
+manager and in turn ask for ordered sequences of these resources according to
+their dependencies or raise exceptions if this is not possible.
+
+Currently, the dependency manager only oversees population initializers for the
+Population manager. In the future its work will expand.
+
+"""
+
+from typing import Sequence, List, Tuple, Callable
+from collections import deque
+
+from vivarium.exceptions import VivariumError
+
+
+class DependencyError(VivariumError):
+    """Error raised when a dependency requirement is violated"""
+    pass
+
+
+class DependencyManager:
+
+    def __init__(self):
+        self.population_initializers = []
+        self.population_initializers_ordered = False
+
+    @property
+    def name(self):
+        return "DependencyManager"
```

Comment:
ok being picky here - do we want to use camel case or snake case by convention for names? I think most of the names that we have so far are snake case (e.g., 'population_manager')?

Reply:
oh yes it should be snake case